pkg/apis/deviceplugin/v1/dsadeviceplugin_types.go

```diff
@@ -28,6 +28,9 @@ type DsaDevicePluginSpec struct {
 	// Image is a container image with DSA device plugin executable.
 	Image string `json:"image,omitempty"`
 
+	// InitImage is an initcontainer image to provision DSA devices with accel-config utility.
+	InitImage string `json:"initImage,omitempty"`
```

Comment:
We should re-think the name.  If we wanted something similar for, e.g., `GpuDevicePlugin`, this entry is already taken. `ProvisioningImage`?

Reply:
IIUC the purpose of `InitImage` is to configure the host (e.g. install tools and drivers) and `ProvisioningImage` configures the devices on the host without touching the host's file system.

I'm thinking if this difference justifies having two separate images for a pod's init containers.

Reply:
@mythi @rojkov

> We should re-think the name. If we wanted something similar for, e.g., GpuDevicePlugin, this entry is already taken. ProvisioningImage?

Regardless of the name it would translate to initcontainers in the deployment?

Implemented either way, it's better to document it clearly. Currently we specify it as "InitImage is an initcontainer image to provision DSA devices with accel-config utility".

As a first step I would simply update it to something like "InitImage / ProvisioningImage is an initcontainer image to configure and enable DSA devices with accel-config utility", enhancing the details if neccessary.

> IIUC the purpose of InitImage is to configure the host (e.g. install tools and drivers) and ProvisioningImage configures the devices on the host without touching the host's file system.

The split in between configuring the host / provisioning SW/HW might be fuzzy (also to understand).

> I'm thinking if this difference justifies having two separate images for a pod's init containers.

My 2 cents: I am fine with ProvisioningImage. I would say there's no right/wrong approach here, I would prefer  a simpler and less confusing solution.

I will update to ProvisioningImage and try to make description less ambiguous.

Reply:
> The split in between configuring the host / provisioning SW/HW might be fuzzy (also to understand).

Can we do both easily with just one image? I'm also fine with one `InitImage` entry per CRD that is tasked to do both if what it needs to do can be customized somehow (e.g. not install some tools but provision only vs do both).

>> I'm thinking if this difference justifies having two separate images for a pod's init containers.
>
> My 2 cents: I am fine with ProvisioningImage. I would say there's no right/wrong approach here,
> I would prefer a simpler and less confusing solution.

+1, the user experience is what matters.

Reply:
`InitImage` is OK